### PR TITLE
CA 234876: make sure Xapi starts after system time is set, clean up startup 

### PR DESCRIPTION
**Pull request by lindig**
The two commits here affect the startup sequence of Xapi:

* To avoid that Xapi reads the system time before it is correct, it is started after the systemd  time-sync.target is reached.

* The old SysV init script `/etc/init.d/xapi` is no longer used. Now it is neither installed nor expected to be installed by the xapi.spec file.

I have slightly re-organised the `After=` constraints in `xapi.service` to make them easier to change manually and with patches.

These changes have been tested by building an RPM, installing it on a DT box and re-booting it.

### CODE REVIEW

File: scripts/xapi.service [hunk 1]
```diff
@@ -1,6 +1,21 @@
 [Unit]
 Description=XenAPI server (XAPI)
-After=syslog.target forkexecd.service xcp-networkd.service message-switch.service v6d.service save-boot-info.service attach-static-vdis.service genptoken.service xcp-rrdd.service xapi-storage-script.service xenopsd-xc.service xenstored.service xcp-networkd.service
```
Comment (mseri) [inline scripts/xapi.service]: Does systemd rearrange the requirements anyway to make it safe to reorder the services in the `Atfer=` or we need to preserve it for any reason?

**Comment by mseri** [comment]
It looks like a good improvement. I have just a minor question before approving it.

File: scripts/xapi.service (hunk 1 above)
Comment (lindig) [inline scripts/xapi.service]: The manual says:

```
This option may be specified more than once, in which case ordering dependencies for
all listed names are created.
```

My understanding is: all After constraints must be met but their ordering is not important. Hence, xapi starts after all of them are reached.

File: scripts/OMakefile [hunk 2]
```diff
@@ -42,10 +42,7 @@ install:
 	$(IDATA) xapi.conf $(DESTDIR)$(XAPICONF)
 	$(IPROG) db.conf.skel $(DESTDIR)$(ETCDIR)/db.conf
 	$(IPROG) rio.db.conf.skel $(DESTDIR)$(ETCDIR)/db.conf.rio
-	mkdir -p $(DESTDIR)/etc/init.d
-	mkdir -p $(DESTDIR)/etc/rc.d/init.d
 	$(IPROG) init.d-xapi $(DESTDIR)$(LIBEXECDIR)/xapi-init
-	$(IPROG) init.d-xapi-compat $(DESTDIR)/etc/init.d/xapi
```
Comment (robhoes) [inline scripts/OMakefile]: Please remove the actual file (scripts/init.d-xapi-compat) as well.

**Comment by lindig** [comment]
I've remove the file - had overlooked that.

**Pull request merged**